driver/reduceextreme_driver.hpp

```diff
@@ -0,0 +1,509 @@
+        std::cerr << "Error ReduceExtremeOp(1-4)\n" << std::endl;
+
+    SetTensorNd(xDesc, in_len, data_type);
+
+    std::vector<int> out_len;
+
+    for(int i = 0; i < in_len.size(); i++)
+    {
+        if(i != dim)
+        {
+            out_len.push_back(in_len[i]);
+        }
+    }
+
+    if(out_len.empty())
+        out_len.push_back(1);
+
+    SetTensorNd(yDesc, out_len, data_type);
+    SetTensorNd(indiceDesc, out_len, indice_data_type);
+
+    return 0;
+}
+
+template <typename Tgpu, typename Tref>
+int ReduceExtremeDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward ReduceExtreme (Default=1)", "int");
+    inflags.AddInputFlag("batchsize", 'n', "21", "Mini-batch size (Default=100)", "int");
+    inflags.AddInputFlag("in_channels", 'c', "500", "Number of Input Channels (Default=3)", "int");
+    inflags.AddInputFlag("in_d", 'D', "0", "Input Depth (Default=0)", "int");
+    inflags.AddInputFlag("in_h", 'H', "0", "Input Height (Default=32)", "int");
+    inflags.AddInputFlag("in_w", 'W', "375", "Input Width (Default=32)", "int");
+    inflags.AddInputFlag(
+        "DimToReduce", 'R', "0", "The indice of the dimensions to be reduced(Default=1)", "int");
+    inflags.AddInputFlag("ReduceExtremeOp",
+                         'O',
+                         "1",
+                         "Reduce Extreme Operation Type (check the enum miopenReduceExtremeOp_t in "
+                         "miopen.h) (Default=1 to Find the the minimum index)",
+                         "int");
+    inflags.AddInputFlag("iter", 'i', "10", "Number of Iterations (Default=10)", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify Each Layer (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "0", "Time Each Layer (Default=0)", "int");
+    inflags.AddInputFlag(
+        "wall", 'w', "0", "Wall-clock Time Each Layer, Requires time == 1 (Default=0)", "int");
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+std::vector<int> ReduceExtremeDriver<Tgpu, Tref>::GetInputTensorLengthsFromCmdLine()
+{
+    int in_n = inflags.GetValueInt("batchsize");
+    int in_c = inflags.GetValueInt("in_channels");
+    int in_w = inflags.GetValueInt("in_w");
+    int in_h = inflags.GetValueInt("in_h");
+    int in_d = inflags.GetValueInt("in_d");
+
+    if((in_n != 0) && (in_c != 0) && (in_d != 0) && (in_h != 0) && (in_w != 0))
+    {
+        return std::vector<int>({in_n, in_c, in_d, in_h, in_w});
+    }
+    else if((in_n != 0) && (in_c != 0) && (in_h != 0) && (in_w != 0))
+    {
+        return std::vector<int>({in_n, in_c, in_h, in_w});
+    }
+    else if((in_n != 0) && (in_c != 0) && (in_w != 0))
+    {
+        return std::vector<int>({in_n, in_c, in_w});
+    }
+    else if((in_n != 0) && (in_w != 0))
+    {
+        return std::vector<int>({in_n, in_w});
+    }
+    else if(in_n != 0)
+    {
+        return std::vector<int>({in_n});
+    }
```

Comment:
[Recommendation] Is it cleaner? Up to you.
```suggestion
    if(in_n != 0)
        if(in_w != 0)
            if(in_c != 0) 
                if(in_h != 0)
                    if(in_d != 0)
                        return std::vector<int>({in_n, in_c, in_d, in_h, in_w});
                    else
                        return std::vector<int>({in_n, in_c, in_h, in_w});
                else
                    return std::vector<int>({in_n, in_c, in_w});
            else
                return std::vector<int>({in_n, in_w});
        else
            return std::vector<int>({in_n});
```

Reply:
The method of receiving tensor input has been changed.

Reply:
[resolved]